addons/mail/static/src/core/common/record.js

```diff
@@ -89,7 +92,17 @@ export class RecordList extends Array {
                 if (name === "length") {
                     return receiver.data.length;
                 }
-                return Reflect.get(target, name, receiver);
+                if (
+                    typeof name === "symbol" ||
+                    Object.keys(target).includes(name) ||
+                    Object.prototype.hasOwnProperty.call(target.constructor.prototype, name)
+                ) {
+                    return Reflect.get(target, name, receiver);
+                } else {
+                    // Attempt an unimplemented array method call
+                    const array = [...receiver];
+                    return array[name].bind(array);
```

Comment:
I don't think this is much of a concern: there's the overhead of proxy for sure, and probably we can't take advantage of memory footprint with generator under-the-hood in case we had a generator record list, which we don't have currently.
At worst, the memory footprint is double during this array call, and there's the proxy overhead that's just a small constant with the rest of the code in this proxy's handlers.